pages/_app.js

```diff
@@ -11,18 +11,17 @@ export default function App({ Component, pageProps }) {
   const [formResults, setFormResults] = useState(0);
   const [randomSurprise, setRandomSurprise] = useState(0);
 
-  // Function to get unique values of the 'activity' key
-  const getUniqueActivities = () => {
-    const uniqueActivities = new Set();
-    dbPlaces.forEach((place) => uniqueActivities.add(place.activity));
-    return Array.from(uniqueActivities);
-  };
-
-  // Function to get unique values of the 'region' key
-  const getUniqueRegions = () => {
-    const uniqueRegions = new Set();
-    dbPlaces.forEach((place) => uniqueRegions.add(place.region));
-    return Array.from(uniqueRegions);
+  // Function to get unique values for a given key
+  const getUniqueValues = (key) => {
+    const uniqueValues = new Set();
+    dbPlaces.forEach((place) => {
+      if (Array.isArray(place[key])) {
```

Comment:
I think this is the only place in your code where you actually have to check for an array